File: benchmarks/compare/recursion/recursion.go
// recursion.go — naive recursive fib + iterative sumTo. Prints "RESULT=<int>".
//
// UNTESTED on the reference machine (no Go toolchain installed); written to be
// correct and to match the other languages' checksum.
package main

import "fmt"

const (
	fibN = 38
	sumN = 50000000
)

func fib(n int32) int64 {
	if n < 2 {
		return int64(n)
	}
	return fib(n-1) + fib(n-2)
}

func sumTo(n int64) int64 {
	var acc int64 = 0
	for i := int64(1); i <= n; i++ {
		acc += i
	}
	return acc
}

func main() {
	f := fib(fibN)
	s := sumTo(sumN)
	result := f*1000000007 + s
	fmt.Printf("RESULT=%d\n", result)
}

File: benchmarks/compare/recursion/recursion.rs
// recursion.rs — naive recursive fib + iterative sumTo. Prints "RESULT=<int>".
const FIB_N: i32 = 42;
const SUM_N: i64 = 50_000_000;

fn fib(n: i32) -> i64 {
    if n < 2 {
        n as i64
    } else {
        fib(n - 1) + fib(n - 2)
    }
}

fn sum_to(n: i64) -> i64 {
    let mut acc: i64 = 0;
    let mut i: i64 = 1;
    while i <= n {
        acc += i;
        i += 1;
    }
    acc
}

fn main() {
    let f = fib(FIB_N);
    let s = sum_to(SUM_N);
    let result = f * 1_000_000_007 + s;
    println!("RESULT={}", result);
}
